backend/src/main/java/com/redhat/cloud/notifications/events/FromCamelHistoryFiller.java

```diff
@@ -70,6 +95,70 @@ void init() {
                 .eventually(() -> messagesProcessedCounter.increment());
     }
 
+    private void reinjectIfNeeded(Map<String, Object> payload) {
+        if (!enableReInject) {
+            return;
+        }
+        if (!payload.containsKey("successful") || !((Boolean) payload.get("successful"))) {
+            String historyId = (String) payload.get("historyId");
+            log.infof("Event with id %s was not successful, resubmitting for further processing", historyId);
+
+            Event event = new Event();
+
+            Uni<Endpoint> endpointUni = notificationResources.getEndpointForHistoryId(historyId);
+            endpointUni.onItem().transform(ep -> {
+
+                // TODO augment with details from Endpoint and original event
+                event.setPayload(payload);
+
+                // Save the original id, as we may need it in the future.
+                Map<String, String> context = new HashMap<>();
+                context.put("original-id", historyId);
+                if (ep != null) { // TODO For the current tests. EP should not be null in real life
+                    context.put("failed-integration", ep.getName());
+                }
+
+                Action action = Action.newBuilder()
+                        .setBundle("platform")
+                        .setApplication("notifications")
+                        .setEventType("integration_failed")
+                        .setAccountId(ep != null ? ep.getAccountId() : "")
+                        .setContext(context)
+                        .setTimestamp(LocalDateTime.now())
+                        .setEvents(Collections.singletonList(event))
+                        .setVersion("v1.1.0")
+                        .setRecipients(Collections.singletonList(
+                                Recipient.newBuilder()
+                                        .setOnlyAdmins(true)
+                                        .setIgnoreUserPreferences(true)
+                                        .build()))
+                        .build();
+
+                String ser = new Encoder().encode(action);
+                // Add the message id in Kafka header for the de-duplicator
+                Message<String> message = buildMessageWithId(ser);
+                emitter.send(message);
+                return Uni.createFrom().voidItem();
+
+            })
+            .subscribe()
+                    .with(x -> x.onItem().invoke(item -> System.out.println(item))); // This triggers the actual work
```

Comment:
`subscribe()` must not be used in runtime code as it will block the Vert.x event loop thread.

Reply:
TBH, this was the only way I could get the processing pipeline to work at all.
Feel free to suggest a better way

Reply:
Done below 😃